### ISSUE TITLE
Very exciting work, but I have a few questions.

### ISSUE THREAD

**Issue by guoqincode** [yu-takagi/StableDiffusionReconstruction#2]
1. Do you already have all the categories in your training set?
2. have you tried testing categories that are not in the training set during testing?
3. Does the Z_T in the paper represent the same outline of the image as the Shape Decoder of Neurips 2020 (Reconstructing Perceptive Images from Brain Activity by Shape-Semantic GAN)?
4. Does c in the paper represent the class of the image (in the case of overfitting) as well as the Semantic Decoder of Neurips 2020 (Reconstructing Perceptive Images from Brain Activity by Shape-Semantic GAN)?

**Comment by yu-takagi** [comment]
@guoqincode  
Thanks you for your questions! Although you have emailed me and I answered via email, I will leave my answer here as well, just in case. Also if you visit [FAQ](https://sites.google.com/view/stablediffusion-with-brain/faq_en) on our project site,  you will find the relevant QA there.
 
1. We are not using fixed category sets to handle images. Instead, we utilize general semantic vectors based on CLIP representations that stem from MC COCO text captions. We also confirmed that using only category labels does reduce the reconstruction accuracy.
2. Our testing data consists of new images with new captions that were not present in the training data.
3. Z is based on the encoder utilized in Stable Diffusion.
4. C is based on the CLIP text encoder.

We understand your concern that our decoding may only recover category-wise results without any visual structure information. In fact, we shared your concern and conducted an examination to determine if decoding using Z+C is better than using just C.  Our results confirm that this is the case (as shown in Figure 5 and the examples in Figure 3). We also confirmed that decoding accuracy decreases when building a semantic decoder using only image category labels. Furthermore, we believe that it would be beneficial to further explore generalizability using non-natural, synthetic stimuli, which is our next step.
